### PR TITLE
Vulnerabilities fixes

### PR DESCRIPTION
**Pull request by asaunier**
Snyk recommends various packages upgrade and patches. This PR aims at addressing those recommendations.

### CODE REVIEW

**Comment by tsauerwein** [comment]
Doesn't Snyk do the PRs itself? And why does the Snyk check still fail?

**Comment by asaunier** [comment]
> Doesn't Snyk do the PRs itself?

No clue.

>  And why does the Snyk check still fail?

Because other actions are required to patch some packages. See
screencapture-snyk-io-test-github-c2corg-v6_ui-f6a3b040346615fa53f29571b6890b22728f6b23-1469610564463

It seems one must run the ``snyk wizard`` https://snyk.io/docs/using-snyk/#wizard
Not sure how I am supposed to install the snyk CLI in the project.

**Comment by tsauerwein** [comment]
> It seems one must run the snyk wizard https://snyk.io/docs/using-snyk/#wizard
> Not sure how I am supposed to install the snyk CLI in the project.

The Snyk CLI was installed with https://github.com/c2corg/v6_ui/pull/490. You can start it with `node_modules/.bin/snyk wizard`.

**Comment by asaunier** [comment]
OK thanks for the tip.
I have run the wizard and basically accepted all its recommendation (upgrade or patch if upgrade is not enough). Not sure everything worked as expected according to the error messages below:
```
 Do you want to revisit your existing policy [y] or only update it [N]? No
Querying vulnerabilities database...
Tested 605 dependencies for known vulnerabilities, found 6 vulnerabilities, 17 vulnerable paths.

? 6 vulnerabilities introduced via openlayers@3.15.1
- info: https://snyk.io/package/npm/openlayers/3.15.1
  Remediation options Upgrade to openlayers@3.17.0

? 2 vulnerabilities introduced in marked
- info: https://snyk.io/package/npm/marked/0.3.5
  Remediation options Patch

? High severity vuln found in minimatch@0.3.0, introduced via openlayers@3.15.1
- desc: Regular Expression Denial of Service
- info: https://snyk.io/vuln/npm:minimatch:20160620
- from: openlayers@3.15.1 > metalsmith@2.1.0 > recursive-readdir@1.3.0 > minimatch@0.3.0
  Remediation options Patch

? 2 vulnerabilities introduced in negotiator
- info: https://snyk.io/package/npm/negotiator/0.4.9
  Remediation options Patch

? 2 vulnerabilities introduced in request
- info: https://snyk.io/package/npm/request/2.61.0
  Remediation options Patch

? 2 vulnerabilities introduced in tar
- info: https://snyk.io/package/npm/tar/1.0.1
  Remediation options Patch

? 2 vulnerabilities introduced in ws
- info: https://snyk.io/package/npm/ws/1.0.1
  Remediation options Patch
? Add `snyk test` to package.json file to fail test on newly disclosed vulnerabilities? Yes
Applying patches...
An unknown error occurred. Please include the trace below when reporting to Snyk:

Error: Failed to patch: the target could not be found.
    at errnoException (child_process.js:988:11)
    at Process.ChildProcess._handle.onexit (child_process.js:779:34)
The patch against "node_modules/openlayers/node_modules/metalsmith/node_modules/recursive-readdir/node_modules/minimatch" (npm:minimatch:20160620) failed.
Email support@snyk.io if this problem persists.
An unknown error occurred. Please include the trace below when reporting to Snyk:

Error: Failed to patch: the target could not be found.
    at errnoException (child_process.js:988:11)
    at Process.ChildProcess._handle.onexit (child_process.js:779:34)
An unknown error occurred. Please include the trace below when reporting to Snyk:

Error: Failed to patch: the target could not be found.
    at errnoException (child_process.js:988:11)
    at Process.ChildProcess._handle.onexit (child_process.js:779:34)
An unknown error occurred. Please include the trace below when reporting to Snyk:

Error: Failed to patch: the target could not be found.
    at errnoException (child_process.js:988:11)
    at Process.ChildProcess._handle.onexit (child_process.js:779:34)
An unknown error occurred. Please include the trace below when reporting to Snyk:

Error: Failed to patch: the target could not be found.
    at errnoException (child_process.js:988:11)
    at Process.ChildProcess._handle.onexit (child_process.js:779:34)
An unknown error occurred. Please include the trace below when reporting to Snyk:

Error: Failed to patch: the target could not be found.
    at errnoException (child_process.js:988:11)
    at Process.ChildProcess._handle.onexit (child_process.js:779:34)
Running `npm update`...
Saving .snyk policy file...
Updating package.json...
Remembering current dependencies for future notifications...

Your .snyk policy file has been successfully updated.
```

See 2c6e797

Among those recommendations, snyk has suggested to upgrade openlayers to 3.17.0!
But then the build fails because
> ./node_modules/.bin/closure-util build build.json c2corg_ui/static/build/build.js
info closure-util Reading build config
info closure-util Getting Closure dependencies
ERR! closure-util Unsatisfied dependency "ol.Extent" in script: /home/alex/Desktop/v6_ui/node_modules/ngeo/src/services/autoprojection.js
Makefile:112: recipe for target 'c2corg_ui/static/build/build.js' failed

Should we upgrade ngeo as well?

**Comment by asaunier** [comment]
@gberaudo helped me upgrading openlayers (to satisfy snyk recommendations) and ngeo. We had to make several changes (see 1e95958) so that closure could build the application.

Unfortunately the application now fails loading because
> Uncaught Error: [$injector:modulerr] http://errors.angularjs.org/1.5.8/$injector/modulerr?p0=app&p1=Error%3A%20%…2Flocalhost%3A6555%2Fnode_modules%2F%2Fangular%2Fangular.min.js%3A20%3A390)

in debug mode:
> GET http://localhost:6555/closure-deps.js => error 500
> base.js:639 goog.require could not find: app.MainController

File: build.json [hunk 1]
```diff
@@ -65,6 +69,6 @@
     "output_wrapper": "(function(){%output%}).call(window);",
     "export_local_property_definitions": true,
     "use_types_for_optimization": true,
-    "manage_closure_dependencies": true
+    "manage_closure_dependencies": false
```
Comment (asaunier) [inline build.json]: If this param is ``true`` closure compiler fails saying that
> ERR! compile node_modules/openlayers/externs/olx.js:1263: ERROR - Bad type annotation. Unknown type ol.control.ScaleLineUnits
ERR! compile  *     units: (ol.control.ScaleLineUnits|string|undefined)}}
ERR! compile                ^
ERR! compile 
ERR! compile node_modules/openlayers/externs/olx.js:1303: ERROR - Bad type annotation. Unknown type ol.control.ScaleLineUnits
ERR! compile  * @type {ol.control.ScaleLineUnits|string|undefined}
ERR! compile           ^
ERR! compile 
ERR! compile node_modules/openlayers/externs/olx.js:1794: ERROR - Bad type annotation. Unknown type ol.format.IGCZ
ERR! compile  * @typedef {{altitudeMode: (ol.format.IGCZ|undefined)}}
ERR! compile                              ^
ERR! compile 
ERR! compile 
ERR! compile node_modules/openlayers/externs/olx.js:1802: ERROR - Bad type annotation. Unknown type ol.format.IGCZ
ERR! compile  * @type {ol.format.IGCZ|undefined}
ERR! compile           ^

Comment (gberaudo) [inline build.json]: Seems an issue with pyramid-closure:
```
  File "/home/gberaudo/dev/c2cv6ui/.build/venv/src/pyramid-closure/pyramid_closure/views.py", line 48, in depsjs
    root, prefix=request.static_url(prefix)))
  File "/home/gberaudo/dev/c2cv6ui/.build/venv/src/pyramid-closure/pyramid_closure/closure/depswriter.py", line 145, in _GetRelativePathToSourceDict
    path_to_source[prefixed_path] = source.Source(source.GetFileContents(path))
  File "/home/gberaudo/dev/c2cv6ui/.build/venv/src/pyramid-closure/pyramid_closure/closure/source.py", line 62, in __init__
    self._ScanSource()
  File "/home/gberaudo/dev/c2cv6ui/.build/venv/src/pyramid-closure/pyramid_closure/closure/source.py", line 105, in _ScanSource
    'Base file should not provide or require namespaces.')
```

**Comment by tsauerwein** [comment]
Ok, I got it to work:

* `angular-ui-date`, which is now a dependency of ngeo, was missing.
* By updating ol3 to https://github.com/openlayers/ol3/pull/5565, I could enable `manage_closure_dependencies` again.

**Comment by tsauerwein** [comment]
And for once Snyk gives green! :+1:

**Comment by asaunier** [comment]
Cool!
Please rebase/fixup the commits and merge, thanks!

**Pull request merged**